cmd/kops/create_cluster.go

```diff
@@ -311,6 +314,8 @@ func NewCmdCreateCluster(f *util.Factory, out io.Writer) *cobra.Command {
 
 	cmd.Flags().StringVar(&options.APILoadBalancerType, "api-loadbalancer-type", options.APILoadBalancerType, "Sets the API loadbalancer type to either 'public' or 'internal'")
 
+	cmd.Flags().StringSliceVar(&options.APILoadBalancerSecurityGroups, "api-loadbalancer-security-groups", options.APILoadBalancerSecurityGroups, "Add precreated additional security groups to the API loadbalancer.")
```

Comment:
I am really trying to not get anymore cli flags and move people toward manifests. Thoughts?

Reply:
@chrislovecnm You are right. I think having too much cli flags is sometimes annoying, but having flexibility to use them simplify some operations. If you create the cluster on demand or some other tool try to create the cluster, that would be more easier. In our case we have a jenkins job create the cluster using the cli completely. We were able to manage the whole deployment  using the cli without editing or modifying any manifest. 

If you have a long running and static setup, the manifests are better in that case. But, if your setup is dynamic and automated, then the cli is better.  

Generally, in kops and kubernetes have the ability to configure and operate the cluster using cli and manifests is a great feature.

Reply:
>  Generally, in kops and kubernetes have the ability to configure and operate the cluster using cli and manifests is a great feature.

Yes until you get too many cli options.  We already have too many.  I love the PR, but another CLI command line is making more of a huge number already ... we do have the override cli option if you want to sneak it in there.  I am trying to find the darn thing.  @justinsb where is the cli option to allow dynamic yaml via cli?

Let me chew on it, I may be ok since it will basically be parity with the other sec group options, but as a whole, I am trying to not have anymore.

>  We were able to manage the whole deployment using the cli without editing or modifying any manifest

I would recommend that you start using templated manifests anyways.  We have dry-run now to build manifest, and templating to do crazy cool stuff with go templating.  

Best practice with my clients is to use YAML.  You will soon outgrow the cli.  Trust me :)

Again thanks

Reply:
@chrislovecnm Thanks for your explanation. I found the kops cluster templating using  toolbox. It is interesting. I mean instead of doing such templating manually using sed or any other tool, using kops toolbox will be more better.

I will update remove the cli arg and restore the cmd gen-cli-docs.